pageserver/src/walredo.rs

```diff
@@ -275,12 +268,11 @@ impl PostgresRedoManager {
                 // Avoid concurrent callers hitting the same issue.
                 // We can't prevent it from happening because we want to enable parallelism.
                 {
-                    let mut guard = self.redo_process.write().unwrap();
-                    match &*guard {
-                        Some(current_field_value) => {
-                            if Arc::ptr_eq(current_field_value, &proc) {
+                    match self.redo_process.get() {
+                        Some(mut guard) => {
+                            if Arc::ptr_eq(&*guard, &proc) {
                                 // We're the first to observe an error from `proc`, it's our job to take it out of rotation.
-                                *guard = None;
+                                drop(guard.take_and_deinit());
```

Comment:
This is Arc::ptr_eq is basically the version check basically which `Layer::evict_blocking` does, but the Arc is already dropped when it is done. Yeah, this looks good. 

`drop` should be done outside the lock while holding the semaphore.

Reply:
I'm not sure I follow you in the above comment. Is there anything you want me to do? If so, please spell it out, otherwise close this conv.

Reply:
I was pointing out that currently the `guard` lives until the end of this block because of an API mistake I forgot (`&mut self` instead of `mut self` on `take_and_deinit`).

As long as the process has no back arrow to `PostgresWalRedoManager` there is no issue, so just resolve this and I'll fix it later.